lib/ClientFactory.php

```diff
@@ -23,7 +23,15 @@
 
 namespace OCA\SharePoint;
 
+use Psr\Log\LoggerInterface;
+
 class ClientFactory {
+	private LoggerInterface $logger;
+
+	public function __construct(LoggerInterface $logger) {
+		$this->logger = $logger;
+	}
```

Comment:
Since this is 26+ only with php8.0+ only you can also do:
```suggestion
	public function __construct(
		private LoggerInterface $logger,
	) {
	}
```

Reply:
>  Since this is 26+ only with php8.0+ only you can also do: 

but it's easier to backport :grimacing:

Reply:
yeah, that's why "Optional"